### ISSUE TITLE
Relay plugin vs GUI Plugins

### ISSUE THREAD

**Issue by wdlkmpx** [weechat/weechat#1096]
I haven't used weechat in a long time.. and to be honest i used it briefly because it's cli app. But i see it has the potential to be ultimate irc client.

Right now i'm using Android and there are 2 weechat apps.. that connect to an actual running weechat ussing the relay interface, like an eggdrop. And i just can't use it like that.

**Comment by wdlkmpx** [comment]
Oops.. continuing.

Have you ever heard about the deadbeef audio player?

Deadbeef  supports a special gui plugin, and you can select the gui plugin.

In Linux it uses gtk2/gtk3, in macOS: cocoaui, in Android it uses a native android gui. It has the potential to run in any platform reliably.

I know creating a gui is a major pain.. but i wonder if weechat is designed to support more than one gui?

**Comment by flashcode** [comment]
Hi,
You can connect to WeeChat with more than one external GUI (via network) at same time. You have the list of them on the download page: https://weechat.org/download/
It's not planned to add a GUI directly in WeeChat itself though, I made the relay plugin with weechat protocol to use external GUI.

**Comment by wdlkmpx** [comment]
https://play.google.com/store/apps/details?id=com.ubergeek42.WeechatAndroid

Yeah but you know, everything is modular, a gui plugin infrastructure would make weechat portable to a native level to every OS known to man.

In fact, the Curses UI or something can be considered a gui for the terminal, now following the same logic, another GUI would be able to connect to the core (instead of the curses UI). Expanding that concept, a user could be able to select a gui in the main config file.

Or, a selection at compile time could happen, compile with curses UI (weechat-cli), with GTK ui (weechat-gtk), etc.... This is the design of several powerful and portable apps.

I'm not saying do it, but consider the advantages of this concept. Of course you don't have to create a gui, just document the API... weechat has excellent documentation.

That way, someone could create a gui for android, iOS, windows (the same as linux), etc... of course dedicated people willing to contribute to the git repo with their efforts....

**Comment by flashcode** [comment]
Yes I planned to make that at the beginning when starting to code WeeChat. But I developed only the Curses interface, and now it would require lot of effort to add other GUI in a modular way.
That's why instead I chose to add the "weechat" protocol in relay plugin, to have external GUI connecting on a running WeeChat via network.
